test/WebJobs.Script.Tests/UtilityTests.cs

```diff
@@ -865,7 +868,78 @@ public void VerifyWorkerIndexingDecisionLogic(bool workerIndexingFeatureFlag, bo
+            bool workerShouldIndex = Utility.CanWorkerIndex(new List<RpcWorkerConfig>() { workerConfig }, testEnv, new FunctionsHostingConfigOptions());
+            Assert.Equal(expected, workerShouldIndex);
+        }
+
+        [Theory]
+        [InlineData(true, true, false, "", true)]
+        [InlineData(true, true, true, "NonApp", true)]
+        [InlineData(true, true, true, "AppName", true)]
+        [InlineData(true, true, false, "NonApp", true)]
+        [InlineData(true, true, false, "AppName", true)]
+        public void VerifyWorkerIndexingFeatureFlagTakesPrecedence(bool workerIndexingFeatureFlag, bool workerIndexingConfigProperty, bool enabledHostingConfig, string disabledHostingConfig, bool expected)
+        {
+            VerifyCanWorkerIndexUtility(workerIndexingFeatureFlag, workerIndexingConfigProperty, enabledHostingConfig, disabledHostingConfig, expected);
+        }
+
+        [Theory]
+        [InlineData(true, false, false, "", false)]
+        [InlineData(true, false, true, "NonApp", false)]
+        [InlineData(true, false, true, "AppName", false)]
+        [InlineData(true, false, false, "NonApp", false)]
+        [InlineData(true, false, false, "AppName", false)]
+        [InlineData(true, true, false, "AppName", true)]
+        public void VerifyWorkerConfigTakesPrecedence(bool workerIndexingFeatureFlag, bool workerIndexingConfigProperty, bool enabledHostingConfig, string disabledHostingConfig, bool expected)
+        {
+            VerifyCanWorkerIndexUtility(workerIndexingFeatureFlag, workerIndexingConfigProperty, enabledHostingConfig, disabledHostingConfig, expected);
+        }
+
+        [Theory]
+        [InlineData(false, true, true, "", true)]
+        [InlineData(false, true, true, "NonApp", true)]
+        [InlineData(false, true, false, "NonApp", false)]
+        [InlineData(false, false, false, "NonApp", false)]
+        public void VerifyStampLevelHostingConfigHonored(bool workerIndexingFeatureFlag, bool workerIndexingConfigProperty, bool enabledHostingConfig, string disabledHostingConfig, bool expected)
+        {
+            VerifyCanWorkerIndexUtility(workerIndexingFeatureFlag, workerIndexingConfigProperty, enabledHostingConfig, disabledHostingConfig, expected);
+        }
+
+        [Theory]
+        [InlineData(false, true, true, "", true)]
+        [InlineData(false, true, true, "NonApp|AppName", false)]
+        [InlineData(false, true, true, "NonApp|AnotherAppName", true)]
+        [InlineData(false, true, true, "nonapp|AppName", false)]
+        [InlineData(false, true, true, "appname", false)]
+        [InlineData(false, true, true, "nonapp|appname", false)]
+        [InlineData(false, true, true, "NonApp|anotherAppname", true)]
+        public void VerifyDisabledAppConfigHonored(bool workerIndexingFeatureFlag, bool workerIndexingConfigProperty, bool enabledHostingConfig, string disabledHostingConfig, bool expected)
```

Comment:
Is DisabledConfig relevant only when FeatureFlag is false? What will happen if featureflag is True for all above cases - will all expected outputs be true?

Reply:
Yes, Feature flag overrides everything except worker indexing configured in worker config. So yes, if feature flag and worker is capable of worker indexing all other config pieces are ignored and result will always be true.